In [ ]:
# Librerías
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
import plotly.express as px

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
# Importamos el DataFrame.
path = "/content/drive/MyDrive/Skillnest/ML/CORES/dataset3.csv"
df = pd.read_csv(path)

In [ ]:
# Visualizamos nuestro DataFrame.
df. info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8807 entries, 0 to 8806
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   show_id       8807 non-null   object
 1   type          8807 non-null   object
 2   title         8807 non-null   object
 3   director      6173 non-null   object
 4   cast          7982 non-null   object
 5   country       7976 non-null   object
 6   date_added    8797 non-null   object
 7   release_year  8807 non-null   int64 
 8   rating        8803 non-null   object
 9   duration      8804 non-null   object
 10  listed_in     8807 non-null   object
 11  description   8807 non-null   object
dtypes: int64(1), object(11)
memory usage: 825.8+ KB


In [ ]:
df.head()

show_id     type                  title         director  \
0      s1    Movie   Dick Johnson Is Dead  Kirsten Johnson   
1      s2  TV Show          Blood & Water              NaN   
2      s3  TV Show              Ganglands  Julien Leclercq   
3      s4  TV Show  Jailbirds New Orleans              NaN   
4      s5  TV Show           Kota Factory              NaN   

                                                cast        country  \
0                                                NaN  United States   
1  Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...   South Africa   
2  Sami Bouajila, Tracy Gotoas, Samuel Jouy, Nabi...            NaN   
3                                                NaN            NaN   
4  Mayur More, Jitendra Kumar, Ranjan Raj, Alam K...          India   

           date_added  release_year rating   duration  \
0  September 25, 2021          2020  PG-13     90 min   
1  September 24, 2021          2021  TV-MA  2 Seasons   
2  September 24, 2021          2021  TV-MA   1 Season   
3  September 24, 2021          2021  TV-MA   1 Season   
4  September 24, 2021          2021  TV-MA  2 Seasons   

                                           listed_in  \
0                                      Documentaries   
1    International TV Shows, TV Dramas, TV Mysteries   
2  Crime TV Shows, International TV Shows, TV Act...   
3                             Docuseries, Reality TV   
4  International TV Shows, Romantic TV Shows, TV ...   

                                         description  
0  As her father nears the end of his life, filmm...  
1  After crossing paths at a party, a Cape Town t...  
2  To protect his family from a powerful drug lor...  
3  Feuds, flirtations and toilet talk go down amo...  
4  In a city of coaching centers known to train I...

In [ ]:
# Verificar valores nulos
df.isna().sum()

show_id            0
type               0
title              0
director        2634
cast             825
country          831
date_added        10
release_year       0
rating             4
duration           3
listed_in          0
description        0
dtype: int64

In [ ]:
# Porcentaje de nulos por columna
porcentaje_nulos = df.isna().mean().sort_values(ascending=False) * 100

# Mostrar en forma de tabla
porcentaje_nulos = porcentaje_nulos.round(2).reset_index()
porcentaje_nulos.columns = ["Columna", "Porcentaje de Nulos"]
porcentaje_nulos

Columna  Porcentaje de Nulos
0       director                29.91
1        country                 9.44
2           cast                 9.37
3     date_added                 0.11
4         rating                 0.05
5       duration                 0.03
6        show_id                 0.00
7           type                 0.00
8          title                 0.00
9   release_year                 0.00
10     listed_in                 0.00
11   description                 0.00

In [ ]:
# Tipo de datos por columna
df.dtypes.value_counts()

object    11
int64      1
Name: count, dtype: int64

In [ ]:
# Cantidad de valores unicos por columna
print("\nCantidad de valores únicos por columna:")
print(df.nunique())



Cantidad de valores únicos por columna:
show_id         8807
type               2
title           8807
director        4528
cast            7692
country          748
date_added      1767
release_year      74
rating            17
duration         220
listed_in        514
description     8775
dtype: int64


In [ ]:
# Verificar filas duplicadas
df.duplicated().sum()

np.int64(0)

no tiene duplicados pero si valores nulos.

In [ ]:
# Descripción de variables numéricas
df.describe().T.round(2)

count     mean   std     min     25%     50%     75%     max
release_year  8807.0  2014.18  8.82  1925.0  2013.0  2017.0  2019.0  2021.0

Cantidad de datos: 8,807 registros (sin valores nulos).
Media: 2014.18 — el año promedio de estreno es 2014.
Desviación estándar: 8.82 — existe una dispersión moderada en los años de estreno.
Mínimo: 1925 — hay contenidos muy antiguos incluidos.


In [ ]:
# Descripción de variables categóricas
df.describe(include="object")

show_id   type   title       director                cast  \
count     8807   8807    8807           6173                7982   
unique    8807      2    8807           4528                7692   
top      s8807  Movie  Zubaan  Rajiv Chilaka  David Attenborough   
freq         1   6131       1             19                  19   

              country       date_added rating  duration  \
count            7976             8797   8803      8804   
unique            748             1767     17       220   
top     United States  January 1, 2020  TV-MA  1 Season   
freq             2818              109   3207      1793   

                           listed_in  \
count                           8807   
unique                           514   
top     Dramas, International Movies   
freq                             362   

                                              description  
count                                                8807  
unique                                               8775  
top     Paranormal activity at a lush, abandoned prope...  
freq                                                    4

Análisis de Variables Categóricas

show_id
Total de registros: 8,807
Valores únicos: 8,807
Cada registro es único (ID único para cada show).

type
Valores únicos: 2 (Movie y TV Show)
Valor más frecuente: Movie con 6,131 registros (≈70% del total).
Predominan las películas sobre los shows de TV.

title
Valores únicos: 8,807
Todos los títulos son únicos (no hay duplicados).

director
Valores únicos: 4,528
Registros con datos no nulos: 6,173 (faltan datos en varios registros)
Director más frecuente: David Attenborough con 19 apariciones.
Gran variedad de directores.

cast
Valores únicos: 7,692
Registros con datos no nulos: 7,982
Actor/actriz más frecuente aparece 19 veces.
Datos de reparto bastante variados pero con valores faltantes.

country
Valores únicos: 748
Registros con datos no nulos: 7,976
País más frecuente: United States con 2,818 registros (≈32%).
Predominan producciones de Estados Unidos, pero el dataset es bastante internacional.

date_added
Valores únicos: 1,767
Registros con datos no nulos: 8,797
Fecha más frecuente: January 1, 2020 con 109 apariciones.

rating
Valores únicos: 17
Registros con datos no nulos: 8,803
Clasificación más frecuente: TV-MA con 3,207 registros (≈36%).

duration
Valores únicos: 220
Registros con datos no nulos: 8,804
Duración más frecuente: 1 Season con 1,793 registros.


listed_in
Valores únicos: 514 categorías (géneros o listas)
Total registros: 8,807
Categoría más frecuente: Dramas, International Movies con 362 apariciones.
Amplia diversidad de géneros y categorías; los dramas internacionales son la categoría más común.

description
Valores únicos: 8,775
Cada descripción es casi única, con muy poca repetición.

In [ ]:
# Gráfico Pie Chart para la distribución por tipo de contenido
fig = px.pie(
    df,
    names="type",
    title="Distribución de Contenido por Tipo (Película vs Serie)",
    hole=0.3
)
fig.update_traces(textinfo="percent+label")
fig.update_layout(title_font_size=18)
fig.show()

El gráfico muestra que la mayoría del contenido en el dataset son películas (69.6%), mientras que el resto son series de TV (30.4%).

In [ ]:
# Histograma de Distribución de Año de Estreno por Tipo de Contenido
fig = px.histogram(
    df,
    x="release_year",
    color="type",
    nbins=7,
    title="Distribución de Año de Estreno por Tipo de Contenido"
)
fig.update_layout(bargap=0.1, title_font_size=18)
fig.show()

El histograma muestra la cantidad de contenidos estrenados por año, separado entre películas y series.

Se puede observar cómo la cantidad de estrenos varía en el tiempo y qué tipo de contenido predomina en cada periodo.

Resumen de Hallazgos
El dataset contiene 12 variables, de las cuales 1 es numérica (release_year) y 11 son categóricas.

Se detectaron valores nulos en 4 variables: director, cast, country y date_added, lo que puede afectar análisis relacionados con producción y procedencia.

Las variables con mayor presencia de outliers no son aplicables directamente, dado que la mayoría de las variables son categóricas; sin embargo, al analizar la duración, se observan valores muy variados que podrían ser considerados extremos si se convierten a formato numérico.

No se pudo realizar el mapa de calor de correlación porque solo existe una variable numérica directa en el dataset (release_year).

Se identificaron desafíos como:
La gran cantidad de variables categóricas con valores faltantes, que requieren limpieza o imputación.
La dificultad para analizar correlaciones cuantitativas debido a la baja cantidad de variables numéricas.

